templates/podDisruptionBudget.cue

```diff
@@ -22,13 +22,16 @@ import (
 
 	spec: {
 		selector: matchLabels: #meta.#LabelSelector
-		if #config[#component].podDisruptionBudget.minAvailable == _|_ && #config.highAvailability.enabled {
+		if #config.highAvailability.enabled &&
+			#config[#component].podDisruptionBudget.minAvailable == _|_ &&
+			#config[#component].podDisruptionBudget.maxUnavailable == _|_ {
 			minAvailable: #config[#component].replicas - 1
 		}
 		if #config[#component].podDisruptionBudget.minAvailable != _|_ {
 			minAvailable: #config[#component].podDisruptionBudget.minAvailable & {uint16 & <=#config[#component].replicas | cfg.#Percent}
 		}
-		if #config[#component].podDisruptionBudget.maxUnavailable != _|_ {
+		if #config[#component].podDisruptionBudget.minAvailable == _|_ &&
+			#config[#component].podDisruptionBudget.maxUnavailable != _|_ {
 			maxUnavailable: #config[#component].podDisruptionBudget.maxUnavailable
```

Comment:
Until to find an better solution, if minAvailable & maxUnavailable is set, maxUnavailable is ignored..